app/controllers/api/base_controller.rb

```diff
@@ -37,8 +37,6 @@ class BaseController < ActionController::API
     rescue_from(ActiveRecord::StatementInvalid) { |e| api_error(:bad_request, e) }
     rescue_from(JSON::ParserError)              { |e| api_error(:bad_request, e) }
     rescue_from(MultiJson::LoadError)           { |e| api_error(:bad_request, e) }
-    rescue_from(MiqException::MiqEVMLoginError) { |e| api_error(:unauthorized, e) }
-    rescue_from(AuthenticationError)            { |e| api_error(:unauthorized, e) }
```

Comment:
I see AuthenticationError raised from within this file so that's ok to remove. I'm not sure about MiqEVMLoginError in the other 2 token based auth.  Should be verified. If that error can be raised, we'd need to rescue MiqException::MiqEVMLoginError for those cases too.

Reply:
I think this is OK because:

1. I see tests that cover the happy/unhappy paths around authenticating with tokens
2. System token authentication does a catch-all rescue, and that specific error inherits from `StandardError`
3. User token authentication doesn't do any actual logging-in as such, it just talks to the token manager